src/libexpr/eval.hh

```diff
@@ -105,18 +102,20 @@ public:
         sDescription, sSelf, sEpsilon, sStartSet, sOperator, sKey, sPath,
         sPrefix,
         sOutputSpecified;
-    Symbol sDerivationNix;
 
     /* If set, force copying files to the Nix store even if they
        already exist there. */
     RepairFlag repair;
 
-    /* The allowed filesystem paths in restricted or pure evaluation
-       mode. */
-    std::optional<PathSet> allowedPaths;
-
     Bindings emptyBindings;
 
+    const ref<FSInputAccessor> rootFS;
+    const ref<MemoryInputAccessor> corepkgsFS;
+
+    const SourcePath derivationInternal;
+
+    std::unordered_map<InputAccessor *, ref<InputAccessor>> inputAccessors;
```

Comment:
This seems like a weird type; as far as I can tell, this is only ever iterated over. Do we really need to have this keyed by raw pointers, as opposed to using something like `std::unordered_set<ref<InputAccessor>>`?